packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+    // Nothing to do -- we don't modify anything.
+    return;
+  }
+
+  private implementsISEPConfiguratorResourceProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!this.implementsIConnectionOptions(value.connection)) { return false; }
+    if (!this.isValidSFRConfig(value.spotFleetRequestConfigurations)) { return false; }
+    if (!this.isValidSpotPluginConfig(value.spotPluginConfigurations)) { return false; }
+    return true;
+  }
+
+  private implementsIConnectionOptions(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!value.hostname || typeof(value.hostname) !== 'string') { return false; }
+    if (!value.port || typeof(value.port) !== 'string') { return false; }
+    const portNum = Number.parseInt(value.port, 10);
+    if (Number.isNaN(portNum) || portNum < 1 || portNum > 65535) { return false; }
+    if (!value.protocol || typeof(value.protocol) !== 'string') { return false; }
+    if (value.protocol !== 'HTTP' && value.protocol !== 'HTTPS') { return false; }
+    if (!this.isSecretArnOrUndefined(value.caCertificate)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.passphrase)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.pfxCertificate)) { return false; }
+    return true;
+  }
+
+  private isSecretArnOrUndefined(value: any): boolean {
+    if (value) {
+      if (typeof(value) !== 'string' || !isSecretArn(value)) { return false; }
+    }
+    return true;
+  }
+
+  private isValidSFRConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+    return true;
+  }
+
+  private isValidSpotPluginConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+
+    return true;
+  }
+
+  private async spotEventPluginRequests(connection: IConnectionOptions): Promise<EventPluginRequests> {
+    return new EventPluginRequests(new DeadlineClient({
+      host: connection.hostname,
+      port: Number.parseInt(connection.port, 10),
+      protocol: connection.protocol,
+      tls: {
+        ca: connection.caCertificate ? await this.readCertificateData(connection.caCertificate) : undefined,
+      },
+    }));
+  }
+
+  private convertToBoolean(input: string): boolean | undefined {
+    try {
+      return JSON.parse(input);
+    } catch(e) {
+      return undefined;
+    }
+  }
+
+  /**
+   * Passing spot fleet request configs into lambda converts all numbers and booleans into strings.
+   * This functions converts these values back to their original type.
+   */
+  private spotFleetRequestToString(spotFleetRequestConfigs: object): string {
+    let convertedSpotFleetRequestConfigs = spotFleetRequestConfigs;
+    for (const [_, sfrConfigs] of Object.entries(convertedSpotFleetRequestConfigs)) {
```

Comment:
I would start with an empty `convertedSpotFleetRequestConfigs` list, iterate over spotFleetRequestConfigs, and build up your configs, adding them to the converted list as you go. Kind of similar to the comment I left below about avoiding mutating objects on the go.

Reply:
Yes, this is much better. Thanks!